tests/chainer_tests/functions_tests/activation_tests/test_softplus.py

```diff
@@ -1,80 +1,56 @@
     'dtype': [numpy.float16, numpy.float32, numpy.float64],
 }))
 @testing.fix_random()
-class TestSoftplus(unittest.TestCase):
+@testing.inject_backend_tests(
+    None,
+    # CPU tests
+    [
+        {},
+        {'use_ideep': True},
+    ]
+    # GPU tests
+    + testing.product({
+        'use_cuda': [True],
+        'cuda_device': [0, 1],
+    })
+    # ChainerX tests
+    + testing.product({
+        'use_chainerx': [True],
+        'chainerx_device': ['native:0', 'cuda:0', 'cuda:1'],
+    })
+)
+class TestSoftplus(testing.FunctionTestCase):
 
     def setUp(self):
-        self.x = numpy.random.uniform(-.5, .5, self.shape).astype(self.dtype)
-        self.gy = numpy.random.uniform(-.5, .5, self.shape).astype(self.dtype)
-        self.ggx = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
         self.beta = numpy.random.uniform(1, 2, ())
         self.check_forward_options = {}
         self.check_backward_options = {}
         if self.dtype == numpy.float16:
             self.check_forward_options = {'atol': 5e-4, 'rtol': 5e-3}
             self.check_backward_options = {'atol': 5e-3, 'rtol': 5e-2}
 
-    def check_forward(self, x_data):
-        x = chainer.Variable(x_data)
-        y = functions.softplus(x, beta=self.beta)
-        x_value = cuda.to_cpu(x_data)
-        y_exp = numpy.log(1 + numpy.exp(self.beta * x_value)) / self.beta
-        self.assertEqual(y.data.dtype, self.dtype)
-        testing.assert_allclose(
-            y_exp, y.data, **self.check_forward_options)
-
-    def check_backward(self, x_data, y_grad):
-        def f(x):
-            return functions.softplus(x, beta=self.beta)
-        gradient_check.check_backward(
-            f, x_data, y_grad,
-            dtype=numpy.float64, **self.check_backward_options)
-
-    def test_forward_cpu(self):
-        self.check_forward(self.x)
-
-    @attr.gpu
-    def test_forward_gpu(self):
-        self.check_forward(cuda.to_gpu(self.x))
-
-    def test_backward_cpu(self):
-        self.check_backward(self.x, self.gy)
-
-    @attr.gpu
-    def test_backward_gpu(self):
-        self.check_backward(cuda.to_gpu(self.x), cuda.to_gpu(self.gy))
-
-    def check_double_backward(self, x_data, y_grad, x_grad_grad,
-                              use_cudnn='always'):
-        def f(x):
-            return functions.softplus(x, beta=self.beta)
-
-        with chainer.using_config('use_cudnn', use_cudnn):
-            gradient_check.check_double_backward(
-                f, x_data, y_grad, x_grad_grad,
-                dtype=numpy.float64, **self.check_backward_options)
+    def generate_inputs(self):
+        x = numpy.random.uniform(-.5, .5, self.shape).astype(self.dtype)
+        return x,
 
-    def test_double_backward_cpu(self):
-        self.check_double_backward(self.x, self.gy, self.ggx)
+    def forward_expected(self, inputs):
+        x, = inputs
+        y = numpy.log(1 + numpy.exp(self.beta * x)) / self.beta
+        if numpy.isscalar(y):
```

Comment:
You can use `force_array` to convert numpy.scalar to numpy.ndarray.
https://github.com/chainer/chainer/blob/2d2956569774a80f3618afe18e6ecb38b097f858/chainer/utils/__init__.py#L32

Reply:
Sure!!

Reply:
Thanks! Made the required changes.